fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -280,13 +277,9 @@ public Response deleteObject() {
 
         try {
             resource().delete();
-
-            try {
-                session.save();
-            } catch (final RepositoryException e) {
-                throw new RepositoryRuntimeException(e);
+            if (!batchService.exists(session.getId(), getUserPrincipal())) {
```

Comment:
The initial implementation of the `FedoraSession` interface held state about such things -- whether it was part of a batch operation, etc, but it was decided to remove that. For a single-node implementation, I can see how holding that state would be very useful and fairly trivial to implement, but if we plan to have a multi-node Fedora implementation at some future point, holding such state becomes really challenging. Which is to say, by pushing that logic into a check on the `BatchService` object, this opens the door to much more robust distributed implementations.

To that I would add that the extensive amount of [reflection](https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/TxAwareSession.java#L65-L82) used in the current implementation to accomplish what you describe seemed weird.